odbc/src/api/functions.rs

```diff
@@ -2595,7 +2595,16 @@ pub unsafe extern "C" fn SQLGetStmtAttrW(
 ///
 #[no_mangle]
 pub unsafe extern "C" fn SQLGetTypeInfo(handle: HStmt, _data_type: SqlDataType) -> SqlReturn {
```

Comment:
Can you please change data_type from SqlDataType to Smallint to match the ODBC function signature and avoid problems if someone provide a dataType value which does not match the list of type in SqlDataType. This is related to [SQL-1151](https://jira.mongodb.org/browse/SQL-1151).
The list is not exhaustive and when the initial input does not match, it will end up with EXT_GUID, which does not match any of our types and therefore the type list returned will be empty, but this is just a by-product of luck.
Just a note that this might affect more code than just this because BsonTypeInfo is using a SqlDataType for sql_type and probably will need to be updated to a Smallint.
I think I would add the SQL types in our definition.rs file and make it an enum with ```#[repr(i16)]
#[derive(Debug, PartialEq, Eq, Clone, Copy, FromPrimitive)]``` so that we can apply a similar approach for converting smallint to SqlType (just a suggestion for the name because it has to not clash with SqlDataType) and can still leverage Enum features.

Reply:
I can make this change, yes. Unless I'm missing something, I think we can remove this function from that ticket once this is done?

Reply:
In implementing, I wonder if a simpler solution could make sense. Specifically, we could compare the `data_type` value to each of our data type's `sql_type.0` (example below). Thus, if we receive a type not in SqlDataType, it will not match with any of our data types, and an empty list of info will be returned. This also limits the scope of changes needed. wdyt?

https://github.com/jchemburkar/mongo-odbc-driver/compare/SQL-1045...jchemburkar:mongo-odbc-driver:SQL-1045-Remove-Enums?expand=1

Reply:
I think this _should_ work if we add the enum and try to match to it in the `SQLGetTypeInfo` function to validate that we received a valid sql type. We can then pass through the smallint to any logic needed by `type_info.rs` (as seen above). My only remaining question would be what SQL types we want in the SqlType enum. Would it just be the 8 or SQL types that correspond to Bson types? that would be my interpretation of the [spec](https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/sql-type-identifiers?view=sql-server-ver16) which seems pretty lenient, but wanted to see if you agree.